stdlib/email/message.pyi

```diff
@@ -146,7 +146,8 @@ class Message(Generic[_HeaderT, _HeaderParamT]):
 class MIMEPart(Message[_HeaderRegistryT, _HeaderRegistryParamT]):
     def __init__(self, policy: Policy | None = None) -> None: ...
     def get_body(self, preferencelist: Sequence[str] = ("related", "html", "plain")) -> MIMEPart[_HeaderRegistryT] | None: ...
-    def iter_attachments(self) -> Iterator[MIMEPart[_HeaderRegistryT]]: ...
+    def attach(self, payload: Self) -> None: ...  # type: ignore[override]
+    def iter_attachments(self) -> Iterator[Self]: ...
```

Comment:
I'm hesitant whether this deserves a comment explaining or linking back to the issue/PR. At least as long as it would fail the test I guess that's fine.

Other than that I would merge once it passes CI (currently fails on 3.12)

Reply:
Makes sense, I've added a comment.